olive/passes/pytorch/slicegpt.py

```diff
@@ -0,0 +1,165 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+# -------------------------------------------------------------------------
+import json
+import logging
+import sys
+from typing import Any, Dict, Union
+
+import torch
+
+from olive.constants import ModelFileFormat
+from olive.data.config import DataConfig
+from olive.hardware.accelerator import AcceleratorSpec
+from olive.model import PyTorchModelHandler
+from olive.model.utils.path_utils import normalize_path_suffix
+from olive.passes import Pass
+from olive.passes.olive_pass import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class SliceGPT(Pass):
+    """Run SliceGPT on a Hugging Face PyTorch model.
+
+    See https://arxiv.org/pdf/2401.15024.pdf for more details on the algorithm.
+
+    This pass only supports PyTorchModelHandler with hf_config.
+    """
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        return {
+            "calibration_data_config": PassConfigParam(
```

Comment:
looks like only the data_name is used from this config. Like we discussed offline previously, wouldn't it be simpler to just have string param called `dataset_name` or something? 

The data config is not used at all but having it here makes the config complicated and also makes it look like any data config is supported.

Reply:
I will follow up on this once I have a discussion with Devang. I think a full data_config keeps the option open for future extension but again it can always be changed in future. Let me follow up after discussion.